core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -130,12 +136,47 @@ public static UnderFileSystemFactory find(String path) {
       }
     }
 
+    // Scan extensions directory on the fly
+    eligibleFactories.addAll(findAllExtensions(path));
```

Comment:
that's a good point. i don't have a good idea of how costly this will be. @apc999 @calvinjia what do you think?